sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/ReadTask.java

```diff
@@ -22,20 +22,20 @@
 import org.apache.spark.annotation.InterfaceStability;
 
 /**
- * A reader factory returned by {@link DataSourceReader#createDataReaderFactories()} and is
+ * A read task returned by {@link DataSourceReader#createReadTasks()} and is
```

Comment:
I still think `ReadTask` is confusing. I were asked by multiple people what is a `ReadTask`, especially when a `Task` is already clearly defined as a unit of execution in Spark. https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/Task.scala#L31-L51

Let us avoid using `Task` here.

Reply:
Ok - how about `ReadTaskDescriptor`? (In that case I think it would be ok to leave the method name as `createReadTasks()`).

Reply:
Hadoop uses the term "input split" for this. Would it be more clear if Spark adopted the same language?

Reply:
`InputSplit` seems a pretty good name for batch, since one split will be processed by one Spark task. How do streaming guys think about it?

Reply:
Okay, one more idea that captures the relationship between `DataReader` and `ReadTask`: What about using `DataReadable`? That's similar to the `Iterable` and `Iterator` relationship that the two classes have.

Reply:
So we want to expose 2 things in the naming:
1. it represents an input RDD partition
2. it creates `DataReader`

I think the first one really needs to be pointed out explicitly, while the second one is not that confusing to create a `DataReader` from a partition. So +1 on `InputPartition`.